File: SampleApp_WebSharperMobileAndroid/bin/html/Scripts/IntelliFactory.WebSharper.Mobile.Android/WebSharper.js
(function()
{
 var global=this;
 var Class,Mobile,__2,localStorage;
 IntelliFactory.WebSharper.Runtime.Declare({IntelliFactory:{WebSharper:{Mobile:{Android:{Mobile:{IStorage:{},IGenericStorage:{}}}}}}});
 Mobile=function()
 {
  return IntelliFactory.WebSharper.Mobile.Android.Mobile;
 };
 (function()
 {
  return IntelliFactory.WebSharper.Mobile.Android;
 });
 (function()
 {
  return IntelliFactory.WebSharper.Mobile;
 });
 (function()
 {
  return IntelliFactory.WebSharper;
 });
 Class=function()
 {
  return IntelliFactory.WebSharper.Runtime.Class;
 };
 (function()
 {
  return IntelliFactory.WebSharper.Runtime;
 });
 __2=function()
 {
  var _this=this;
  var _,__1,c,matchValue;
  matchValue={$:0};
  if(matchValue.$==1)
   {
    c=matchValue.$0;
    _=[];
    __1=void c.apply(_this,_);
   }
  __1;
 };
 localStorage=function()
 {
  return window.localStorage;
 };
 (Mobile()).IGenericStorage=(Class())(__2,null,{get_Item:function(v)
 {
  var _,__3,__4,__5,__6;
  _=localStorage();
  __4=_.getItem(v);
  __3=function(arg00)
  {
   var __1;
   __1=JSON;
   return JSON.parse.call(__1,arg00);
  };
  __6=__3(__4);
  __5=function(value)
  {
   return value;
  };
  return __5(__6);
 },set_Item:function(k,v)
 {
  var _,__1,__3;
  __1=localStorage();
  _=JSON;
  __3=JSON.stringify.call(_,v);
  return __1.setItem(k,__3);
 }});
 (Mobile()).IStorage=(Class())(__2,null,{get_Item:function(v)
 {
  var _;
  _=localStorage();
  return _.getItem(v);
 },set_Item:function(k,v)
 {
  var _;
  _=localStorage();
  return _.setItem(k,v);
 }});
 (Mobile()).GetLocation=function()
 {
  var _,s;
  _=websharperBridge;
  s=websharperBridge.location.call(_);
  return eval(""+s);
 };
 (Mobile()).GetAcceleration=function()
 {
  var _,s;
  _=websharperBridge;
  s=websharperBridge.acceleration.call(_);
  return eval(""+s);
 };
 (Mobile()).Storage=new(Mobile()).IStorage();
 (Mobile()).GenericStorage=new(Mobile()).IGenericStorage();
}());

File: if-ws-mobile-android/WebSharper.fs
// $begin{copyright}
//
// This file is confidential and proprietary.
//
// Copyright (c) IntelliFactory, 2004-2011.
//
// All rights reserved.  Reproduction or use in whole or in part is
// prohibited without the written consent of the copyright holder.
//-----------------------------------------------------------------
// $end{copyright}

module IntelliFactory.WebSharper.Mobile.Android

open IntelliFactory.WebSharper
open IntelliFactory.WebSharper.Html
open IntelliFactory.WebSharper.Html5
open IntelliFactory.WebSharper.Json
open IntelliFactory.WebSharper.Mobile

[<assembly: System.Web.UI.WebResource("android.websharperBridge.serverLocation.js", "text/javascript", PerformSubstitution = true)>]
do ()
type private AndroidServerLocationResource() =
    inherit Resources.BaseResource("android.websharperBridge.serverLocation.js")

[<Require(typeof<AndroidServerLocationResource>)>]
module private Ajax =

    [<JavaScript>]
    let private callbacks = new System.Collections.Generic.Dictionary<int,obj>()

    [<JavaScript>]
    let private failureCallbacks = new System.Collections.Generic.Dictionary<int,obj>()

    [<JavaScript>]
    let mutable private callbackCounter = 0

    [<JavaScript>]
    let setCallbacks success failure =
        let cbc = callbackCounter
        callbackCounter <- callbackCounter + 1
        callbacks.[cbc] <- success
        failureCallbacks.[cbc] <- failure
        "IntelliFactory.WebSharper.Mobile.Android.Ajax.callbacks.get_Item(" + cbc.ToString() + ")",
            "IntelliFactory.WebSharper.Mobile.Android.Ajax.failureCallbacks.get_Item(" + cbc.ToString() + ")"

[<Require(typeof<AndroidServerLocationResource>)>]
type private AndroidMobileProvider [<JavaScript>] () =
    
    [<Inline "websharperBridge.alert($s)">]
    let bAlert (s : string) : unit = X

    [<Inline "websharperBridge.log($s)">]
    let bLog (s : string) : unit = X

    [<Inline "eval('' + $s)">]
    let eval (s : string) = X
    
    [<Inline "websharperBridge.location()">]
    let bLocation () : string = X

    [<Inline "websharperBridge.acceleration()">]
    let bAcceleration () : string = X

    [<Inline "websharperBridge.camera($width, $height, $success, $fail)">]
    let bPhotoFromCamera (width : int) (height : int) (success : string) (fail : string) : unit = X

    [<JavaScript>]
    let photoFromCamera (width, height) (success, fail, _) =
        let success, fail = Ajax.setCallbacks (success << fun s -> "" + s) fail
        bPhotoFromCamera width height success fail

    interface IMobileProvider with

        [<JavaScript>]
        override this.Alert s = bAlert s

        [<JavaScript>]
        override this.Log s = bLog s

        [<JavaScript>]
        override this.GetLocation () = bLocation() |> eval

        [<JavaScript>]
        override this.GetAcceleration () = bAcceleration() |> eval
        
        [<JavaScript>]
        override this.GetPhotoFromCamera () = //(width, height)
            Async.FromContinuations (photoFromCamera (0, 0))

        [<JavaScript>]
        override this.StorageLoad k =
            try
                Window.Self.LocalStorage.GetItem k
                |> function
                    | x when x = null -> ""
                    | x -> x
            with _ -> ""

        [<JavaScript>]
        override this.StorageStore k v = Window.Self.LocalStorage.SetItem (k, v)

        [<JavaScript>]
        override this.JsonStorageLoad k =
            (this :> IMobileProvider).StorageLoad k
            |> function
                | "" -> "[]"
                | x -> x
            |> Json.Parse
            |> unbox

        [<JavaScript>]
        override this.JsonStorageStore k v = (this :> IMobileProvider).StorageStore k (Json.Stringify v)

module private ProviderHolder =
    [<JavaScript>]
    let provider = AndroidMobileProvider()

[<Inline "android_init()">]
let private androidInit () : unit = X

[<JavaScript>]
let EnableAndroidSupport () =
    ignore Mobile
    ignore IntelliFactory.WebSharper.Remoting.Config.EndPoint
    androidInit()
